src/netbase.h

```diff
@@ -246,4 +246,17 @@ void InterruptSocks5(bool interrupt);
  */
 bool Socks5(const std::string& strDest, uint16_t port, const ProxyCredentials* auth, const Sock& socket);
 
+/**
+ * Determine if a port is "bad" from the perspective of attempting to connect
+ * to a node on that port.
+ * See
+ * https://fetch.spec.whatwg.org/#port-blocking
```

Comment:
I was wondering which of the 3 sources the used list is based on, but they seem to be identical with the exception of port 0, which is considered bad by [Mozilla](https://hg.mozilla.org/mozilla-central/file/tip/netwerk/base/nsIOService.cpp#l1389). Should we add port 0 too?

Reply:
I think i2p uses port 0, so we'd need to make an exception for that

Reply:
TCP port 0 does not exist. It also doesn't appear in the Mozilla list; it's just a terminator to indicate "end of list" there.

@benthecarman Right; this list should only apply to IPv4/IPv6.

Reply:
All 3 sources agree on which ports are bad (which is good, pun intended). Can ignore port 0, that is something like opening connection to port 70000 or to IP address `1.2.3.500`.

Changed the logic to apply only to clearnet (IPv4 and IPv6).